datafusion/src/execution/dataframe_impl.rs

```diff
@@ -231,6 +231,29 @@ impl DataFrame for DataFrameImpl {
                 .build()?,
         )))
     }
+
+    fn intersect(&self, dataframe: Arc<dyn DataFrame>) -> Result<Arc<dyn DataFrame>> {
+        let left_plan = self.to_logical_plan();
+        let right_plan = dataframe.to_logical_plan();
+        let join_keys = left_plan
+            .schema()
+            .fields()
+            .iter()
+            .zip(right_plan.schema().fields().iter())
+            .map(|(left_field, right_field)| {
+                (
+                    (Column::from_name(left_field.name())),
+                    (Column::from_name(right_field.name())),
+                )
+            })
+            .unzip();
+        Ok(Arc::new(DataFrameImpl::new(
+            self.ctx_state.clone(),
+            &LogicalPlanBuilder::from(left_plan)
+                .join_detailed(&right_plan, JoinType::Semi, join_keys, true)?
+                .build()?,
```

Comment:
I think once we merge https://github.com/apache/arrow-datafusion/pull/1259 this could just call `LogicalPlanBuilder::intersect`

Reply:
yes！

Reply:
Yes, once #1259 is merged, I'll quickly fix the ticket and #1261. 😊